enhancements/olm/service-account-deprecation.md

```diff
@@ -0,0 +1,262 @@
+---
+title: serviceaccount-field-deprecation
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@everettraven"
+  - "@grokspawn"
+  - "@trgeiger"
+  - "@joelanford"
+approvers:
+  - "@everettraven"
+  - "@trgeiger"
+  - "@grokspawn"
+creation-date: 2025-10-08
+last-updated: 2025-10-15
+status: implementable
+---
+
+# Service Account Field Deprecation
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Deprecate the `.spec.serviceAccount` field from the ClusterExtension API in the operator-controller. This field was originally introduced to enforce least privilege by requiring users to provide a ServiceAccount with the necessary RBAC permissions to manage extension content. This proposal removes that requirement and simplifies controller logic and behavior.
+
+**Note**: This deprecation is an interim step. While the intent is to eventually remove the `.spec.serviceAccount` field, we will keep it as optional and mark it as deprecated during this transition period spanning multiple releases. This approach allows users time to adapt without disruption before the field is fully removed.
+
+## Motivation
+
+The original intent of `.spec.serviceAccount` was to support a least-privilege model by allowing users to provide custom `ServiceAccount` with fine-grained permissions. In practice, this design introduced considerable operational and technical complexity, including:
+
+- Token acquisition, rotation, and error handling  
+- Custom `rest.Config` generation and dynamic HTTP clients  
+- Token expiration edge cases 
+
+Given the limited benefit and high complexity of this approach, we propose simplifying the model:
```

Comment:
It would be helpful if you could clarify what the "limited benefit" here is.

You start this section talking about how the intention was originally for following a least-privilege principle which, on the surface, sounds like it would actually be a pretty big benefit.

After all, looking at the documentation for OLMv1 it seems like "secure by default" is a core tenant: https://github.com/operator-framework/operator-controller/blob/292c0dbc236747e57f77fd914830a865e87fad6b/docs/project/olmv1_design_decisions.md?plain=1#L179-L186

As a reviewer, I'd like to better understand:
- Why this tenant was originally seen as a win for end users
- Why this tenant turned out to _not_ be a win for end users